test/quantization/test_quantize_script.py

```diff
@@ -2244,10 +2240,9 @@ def __init__(self):
             def forward(self, x):
                 return self.fc(x)
 
-        data = torch.rand((1, 5), dtype=torch.float)
         qconfig_dict = {'': default_dynamic_qconfig}
         model = torch.jit.script(M()).eval()
-        model = quantize_dynamic_script(model, qconfig_dict, data)
+        model = quantize_dynamic_script(model, qconfig_dict)
```

Comment:
Is there a test for shared weights?
 For example, if I have a module of the form:
```python
class DynamicModel(torch.nn.Module):
    def __init__(self):
        super(DynamicModel, self).__init__()
        self.weight  = torch.rand(10,10)
        self.mod1 = myMod(self.weight)

 def forward(self, x):
       y = self.mod1(x)
       z = torch.nn.functional.linear(y, self.weight)
       return z

class myMod(torch.nn.Module):
    def __init__(wt):
        super(myMod, self).__init__()
        self.linear = nn.Linear(10, 10)
        self.linear.weight = wt

 def forward(self, x):
       return self.linear(x)
      
```
Can we handle cases where only myMod is quantized (has a qconfig) and the reverse (only MyMod is in float?)